src/features/common/Layout/ProjectPropsContext.tsx

```diff
@@ -1,124 +1,246 @@
   selectedSite: 0,
-  setSelectedSite: (value: number) => {},
+  setSelectedSite: () => {},
   siteExists: false,
-  setsiteExists: (value: boolean) => {},
-  isMobile: false,
+  setsiteExists: () => {},
   infoRef: {} || null,
   exploreContainerRef: {} || null,
   exploreExpanded: false,
-  setExploreExpanded: (value: boolean) => {},
+  setExploreExpanded: () => {},
   exploreForests: false,
-  setExploreForests: (value: boolean) => {},
+  setExploreForests: () => {},
   explorePotential: false,
-  setExplorePotential: (value: boolean) => {},
+  setExplorePotential: () => {},
   exploreDeforestation: false,
-  setExploreDeforestation: (value: boolean) => {},
+  setExploreDeforestation: () => {},
   explorePlanted: false,
-  setExplorePlanted: (value: boolean) => {},
+  setExplorePlanted: () => {},
   infoExpanded: null || '',
-  setInfoExpanded: (value: string) => {},
+  setInfoExpanded: () => {},
   openModal: false,
-  setModalOpen: (value: boolean) => {},
+  setModalOpen: () => {},
   viewport: {},
-  setViewPort: (value: {}) => {},
-  setExploreProjects: (value: boolean) => {},
+  setViewPort: () => {},
   mapState: {},
-  setMapState: (value: {}) => {},
+  setMapState: () => {},
   exploreProjects: true,
+  setExploreProjects: () => {},
   loaded: false,
-  setLoaded: (value: boolean) => {},
+  setLoaded: () => {},
   mapRef: {},
   defaultMapCenter: [],
   defaultZoom: 1.4,
   layersSettings: {},
-  setLayersSettings: (value: {}) => {},
+  setLayersSettings: () => {},
   selectedMode: 'location',
-  setSelectedMode: (value: string) => {},
+  setSelectedMode: () => {},
   rasterData: {
     evi: '',
     imagery: {},
   },
-  setRasterData: (value: {}) => {},
-  plantLocations: [],
-  setPlantLocations: (value: []) => {},
+  setRasterData: () => {},
   selectedPl: {},
-  setSelectedPl: (value: {}) => {},
+  setSelectedPl: () => {},
   zoomLevel: 1,
-  setZoomLevel: (value: number) => {},
+  setZoomLevel: () => {},
   satellite: false,
-  setSatellite: (value: boolean) => {},
+  setSatellite: () => {},
   plIds: null || [],
-  setPlIds: (value: []) => {},
+  setPlIds: () => {},
   hoveredPl: null || '',
-  setHoveredPl: (value: {}) => {},
+  setHoveredPl: () => {},
   isPolygonMenuOpen: false,
-  setIsPolygonMenuOpen: (value: boolean) => {},
+  setIsPolygonMenuOpen: () => {},
   siteViewPort: null || {},
-  setSiteViewPort: (value: {}) => {},
+  setSiteViewPort: () => {},
   filteredProjects: null || [],
-  setFilteredProjects: (value: []) => {},
+  setFilteredProjects: () => {},
   filtersOpen: false,
-  setFilterOpen: (value: boolean) => {},
+  setFilterOpen: () => {},
   purpose: 'trees',
-  setPurpose: (value: string) => {},
+  setPurpose: () => {},
   plantLocationsLoaded: false,
-  setPlantLocationsLoaded: (value: boolean) => {},
-});
+  setPlantLocationsLoaded: () => {},
+  isMobile: false,
+  setIsMobile: () => {},
+};
+
+export const ProjectPropsContext =
+  React.createContext<ProjectContextInterface>(defaultState);
```

Comment:
In my VSC I still get displayed this error:

```
Argument of type '{ project: {} | null; setProject: () => void; projects: never[]; setProjects: () => void; plantLocations: never[]; setPlantLocations: () => void; showSingleProject: boolean; setShowSingleProject: () => void; ... 66 more ...; setIsMobile: () => void; }' is not assignable to parameter of type 'ProjectContextInterface'.
  Types of property 'project' are incompatible.
    Type '{} | null' is not assignable to type 'Project | null'.
      Type '{}' is missing the following properties from type 'Project': id, _scope, allowDonations, classification, and 44 more.ts(2345)
```
I am not yet experienced with Typescript typing, so maybe you have an idea about this. I think I used the default settings for Typescript checks in Visual Studio Code.

Reply:
I changes the types and default values. Please have a look. If this is wrong, please revert my commit.

Reply:
I again have problems as you used uppercase characters for branch names `Feature/typescript_projectPropContext`. Can we please not used uppercase characters in branch names?

Reply:
I am not sure. but, whenever I create branch using lowercase character, I am unable to push the changes into remote repo